tcp/server_test.go

```diff
@@ -253,20 +262,24 @@ func TestServer_KeepAliveMonitor(t *testing.T) {
 
 	cc, err := tcp.Dial(
 		ld.Addr().String(),
-		tcp.WithInactivityMonitor(time.Millisecond*10, func(cc inactivity.ClientConn) {
+		tcp.WithGoPool(func(f func()) error {
 			time.Sleep(time.Millisecond * 500)
+			f()
+			return nil
 		}),
 	)
+
 	require.NoError(t, err)
 	checkCloseWg.Add(1)
 	cc.AddOnClose(func() {
 		checkCloseWg.Done()
 	})
 
 	// send ping to create serverside connection
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	ctx, cancel = context.WithTimeout(context.Background(), time.Second)
```

Comment:
Doesn't this overwrite the defered `cancel` on line 232? (I always forget how exactly does defer behave :zipper_mouth_face: )